analysis/data-fetch.R

```diff
@@ -0,0 +1,30 @@
+
+library(tidyverse)
+library(fs)
+
+#download_dir <- tempdir()
+
+url_pattern <- "http://www2.susep.gov.br/redarq.asp?arq=Autoseg{year}{half}%2ezip"
+
+download_params <- tibble(year = 2012, half = c("B"))
+
+download_urls <- glue::glue_data(.x = download_params, url_pattern)
+
+download_dir <- path(here::here(), "external_data") 
+
+if(!dir.exists(download_dir)) {
+  
+  dir.create(download_dir)
+
+}
+
+download_file <- function(x, dest_dir) {
+  
```

Comment:
We might want to check to see if file is downloaded already.

Reply:
Makes sense. I will add tonight, if I can. Also, should zip file be left in folder or deleted? Should user have the option?

Reply:
I think an option makes sense (probably default to remove). The file exist check should also take into account uncompressed folders without the zip.

Reply:
Added function ```download_data``` to pricingtutorial package and modified data-fetch.R appropriately.